__tests__/github.test.js

```diff
@@ -0,0 +1,46 @@
+const pool = require('../lib/utils/pool');
+const setup = require('../data/setup');
+const request = require('supertest');
+const app = require('../lib/app');
+const GithubUser = require('../lib/models/GithubUser');
+
+jest.mock('../lib/utils/github');
+
+describe('Gitty routes', () => {
+  beforeEach(() => {
+    return setup(pool);
+  });
+
+  afterAll(() => {
+    pool.end();
+  });
+
+  it('redirects to the github oauth page upon login', async () => {
+    const req = await request(app).get('/api/v1/github/login');
+
+    expect(req.header.location).toMatch(
+      /https:\/\/github.com\/login\/oauth\/authorize\?client_id=[\w\d]+&scope=user&redirect_uri=http:\/\/localhost:7890\/api\/v1\/github\/login\/callback/i
+    );
+  });
+
+  it('should login and redirect to posts', async () => {
+    const req = await request
+      .agent(app)
+      .get('/api/v1/github/login/callback?code=42')
+      // check for res.redirects to see if it directed us back to the posts route
+      .redirects(1);
+    expect(req.redirects[0]).toEqual(expect.stringContaining('/api/v1/posts'));
+  });
+
+  it('logs a user out through a delete route', async () => {
+    await GithubUser.createUser({
```

Comment:
Instead of using GthubUser.createUser here, just hit your login/callback route. It'll take care of the user for us.